data-access-layer/cf/TokenIssuer.js

```diff
@@ -63,6 +63,38 @@ class TokenIssuer {
       .catch(() => this.login())
       .then(result => this.updateTokenInfo(result).accessToken);
   }
+
+  scheduleNextRequestAccessToken(clientId, clientSecret) {
+    const delay = this.tokenInfo.accessTokenExpiresIn - 15;
+    logger.info('delay: ', delay);
+    if (delay > 0 && delay < 2147483647) {
+      this.clearTimeoutObject();
+      logger.info('scheduling next request for access token after delay:', delay * 1000);
+      this.timeoutObject = setTimeout(() => {
+        logger.info('requesting new access token with client id: ', clientId);
+        return this.uaa.accessWithClientCredentials(clientId, clientSecret)
+          .then(token => {
+            this.tokenInfo.update(token);
+            this.scheduleNextRequestAccessToken(clientId, clientSecret);
+          })
+          .catch(err => logger.error(err.message));
+      }, delay * 1000);
+    }
+  }
+
+  getAccessTokenBoshUAA(clientId, clientSecret) {
+    if (!this.tokenInfo.accessTokenExpiresSoon) {
+      logger.info('reusing access token.');
```

Comment:
better to make this debug , since it can fill up the logs for every bosh access.